File: src/main/java/net/cjsah/plugin/mcspider/McSpider.java
package net.cjsah.plugin.mcspider;

import net.cjsah.console.Console;
import net.cjsah.console.Permission;
import net.cjsah.console.command.Command;
import net.cjsah.console.command.CommandManager;
import net.cjsah.console.command.argument.IntArgument;
import net.cjsah.console.command.argument.StringArgument;
import net.cjsah.console.plugin.Plugin;

public class McSpider extends Plugin {
    @Override
    protected void onPluginLoad() {
        Config.INSTANCE.init(this);
    }

    @Override
    public void onBotStarted() {
        CommandManager.register(dispatcher ->
                dispatcher.register(CommandManager.literal("mcv").requires(source -> source.hasPermission(Permission.HELPER)).executes("获取最新mc版本", context -> {
                    context.getSource().sendFeedBack(GetVersion.INSTANCE.getVersion(Console.INSTANCE.getBot()));
                    return Command.SUCCESSFUL;
                }))
        );

        CommandManager.register(dispatcher ->
                dispatcher.register(CommandManager.literal("server").then(CommandManager.argument("address", StringArgument.string()).requires(source -> source.hasPermission(Permission.HELPER)).executes("获取某服务器的信息", context -> {
                    context.getSource().sendFeedBack(GetServer.INSTANCE.getServer(StringArgument.getString(context, "address")));
                    return Command.SUCCESSFUL;
                })))
        );

        CommandManager.register(dispatcher ->
                dispatcher.register(CommandManager.literal("mc").then(CommandManager.argument("value", IntArgument.integer(0)).requires(source -> source.hasPermission(Permission.HELPER)).executes("获取 mc bugs", context -> {
                    context.getSource().sendFeedBack(GetBugs.INSTANCE.getBugs(IntArgument.getInteger(context, "value")));
                    return Command.SUCCESSFUL;
                })))
        );

        GetVersion.INSTANCE.scope();
    }

    @Override
    public void onBotStopped() {

    }

    @Override
    public void onPluginUnload() {

    }
}

File: src/main/kotlin/net/cjsah/plugin/mcspider/McSpider.kt
package net.cjsah.plugin.mcspider

import net.cjsah.console.command.Command
import net.cjsah.console.command.CommandManager
import net.cjsah.console.command.argument.IntArgument
import net.cjsah.console.command.argument.StringArgument
import net.cjsah.console.command.source.CommandSource
import net.cjsah.console.plugin.Plugin
import net.cjsah.plugin.mcspider.Utils.checkVersion
import net.cjsah.plugin.mcspider.Utils.getBugs
import net.cjsah.plugin.mcspider.Utils.getServer
import net.cjsah.plugin.mcspider.Utils.getVersion
import java.util.Timer
import java.util.TimerTask

class McSpider : Plugin() {

    private val timer = Timer()

    override fun onPluginLoad() {
        Config.init(this)
    }

    override fun onBotStarted() {
        CommandManager.register { dispatcher ->
            dispatcher.register(CommandManager.literal("mcv").requires { source ->
                source.canUse(this)
            }.executes("获取最新mc版本") { context ->
                context.getSource().sendFeedBack(getVersion())
                Command.SUCCESSFUL
            })
        }

        CommandManager.register { dispatcher ->
            dispatcher.register(CommandManager.literal("server")
                .then(CommandManager.argument("address", StringArgument.string()).requires { source ->
                    source.canUse(this)
                }.executes("获取某服务器的信息") { context ->
                    context.getSource().sendFeedBack(getServer(StringArgument.getString(context, "address")))
                    Command.SUCCESSFUL
                })
            )
        }

        CommandManager.register { dispatcher ->
            dispatcher.register(CommandManager.literal("mc")
                .then(CommandManager.argument("value", IntArgument.integer(0))
                    .requires { source: CommandSource<*> ->
                        source.canUse(this)
                    }.executes("获取 mc bugs") { context ->
                        context.getSource().sendFeedBack(getBugs(IntArgument.getInteger(context, "value")))
                        Command.SUCCESSFUL
                    }
                )
            )
        }

        timer.schedule(object : TimerTask() {
            override fun run() {
                checkVersion(this@McSpider)
            }
        }, 1, 180000)

    }

    override fun onBotStopped() {
        timer.cancel()
    }

    override fun onPluginUnload() {
    }
}